SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp

```diff
@@ -414,10 +444,27 @@ void USpatialSender::SendRPC(TSharedRef<FPendingRPCParams> Params)
 		// Target object was destroyed before the RPC could be (re)sent
 		return;
 	}
-
 	UObject* TargetObject = Params->TargetObject.Get();
+
+	if (USpatialActorChannel* Channel = NetDriver->GetOrCreateSpatialActorChannel(TargetObject))
+	{
+		if (Channel->bCreatingNewEntity)
+		{
+			check(PackageMap->GetUnrealObjectRefFromObject(TargetObject) != FUnrealObjectRef::UNRESOLVED_OBJECT_REF);
```

Comment:
Can we also be explicit here that we expect this to only happen on servers and only for client and multicast RPCs?
Also, add a comment + TODO for doing additional work for multicast RPCs (because currently they will only be executed on the owning client, if any).

Reply:
Added requested checks.